app/helpers/admin/site_maps_helper.rb

```diff
@@ -3,12 +3,26 @@
 module Admin::SiteMapsHelper
   include Admin::AdminHelper
 
+  def sitemap_pages
+    @sitemap_pages ||= Page.where(:_type.in => %w(Page SetPage)).entries
```

Comment:
Could you get all the `SetPage`s first, and then the sitemap pages query would be:

```ruby
Page.where(:page_id.nin => set_pages.map(&:id))
```

?